### DISCUSSION TITLE
How to convert back and forth between Point3d and Cartographic

### DISCUSSION THREAD

**Discussion by JohnMajerle** [iTwin/itwinjs-core#5312] [Q&A]
See below test code snippet.  I need to be able to convert back and forth between a Point3d and a Cartographic.  When I execute the code snippet I find that center2 does not have the same  values as center1.

What is the proper way to do perform the conversion?

**Code Snippet**

            let center1 = viewPort.getFrustum().toRange().center;
            let cartographic = viewPort.iModel.spatialToCartographicFromEcef(center1);
            let center2 = cartographic.toEcef();

**Comment by pmconne** [comment]
The `fromEcef` method uses the iModel's ECEF transform, which is imprecise over large distances. If the iModel defines a geocoordinate system (GCS), `IModelConnection.spatialToCartographicFromGcs` will use the GCS to do more precise conversions.

What you really want is [IModelConnection.spatialToCartgoraphic](https://www.itwinjs.org/reference/core-frontend/imodelconnection/imodelconnection/spatialtocartographic/), which will decide which of the other two functions to call based on whether the iModel defines a GCS or not.
The inverse is `IModelConnection.cartographicToSpatial`.

Note: these methods are asynchronous. If you have multiple points to convert, prefer to batch them up using `Promise.all` rather than `await`ing each conversion in sequence. We should probably add versions of them that accept an array of points.
